feat: Selective CSP header stripping from HTTPResponse (#26483)
cypress-io/cypress: cli/CHANGELOG.md

<!-- See the ../guides/writing-the-cypress-changelog.md for details on writing the changelog. -->
## 12.14.0

_Released 06/06/2023 (PENDING)_

**Features:**

> - Cypress now has a targeted Content-Security-Policy and Content-Security-Policy-Report-Only header directive allow list via the [`experimentalCspAllowList`](https://docs.cypress.io/guides/references/configuration#Experimental-Csp-Allow-List) configuration option. Addresses [#1030](https://github.com/cypress-io/cypress/issues/1030).

Reviewer: I think we might need the 'Addressed in' for features to make the changelog job happy (sorry if I said otherwise 🥴 )
```suggestion
- Cypress now has a targeted Content-Security-Policy and Content-Security-Policy-Report-Only header directive allow list via the [`experimentalCspAllowList`](https://docs.cypress.io/guides/references/configuration#Experimental-Csp-Allow-List) configuration option. Addresses [#1030](https://github.com/cypress-io/cypress/issues/1030). Addressed in [#26483](https://github.com/cypress-io/cypress/pull/26483)
```

> - Cypress now has a targeted Content-Security-Policy and Content-Security-Policy-Report-Only header directive allow list via the [`experimentalCspAllowList`](https://docs.cypress.io/guides/references/configuration#experimentalCspAllowList) config option. Addresses [#1030](https://github.com/cypress-io/cypress/issues/1030). Addressed in [#26483](https://github.com/cypress-io/cypress/pull/26483).

Reviewer: ```suggestion
- Cypress now has a targeted Content-Security-Policy and Content-Security-Policy-Report-Only header directive allow list via the [`experimentalCspAllowList`](https://docs.cypress.io/guides/references/configuration#experimentalCspAllowList) configuration option. Addresses [#1030](https://github.com/cypress-io/cypress/issues/1030).
```

Author: @AtofStryker I removed the "Addressed" item, but the Semantic PR automation yelled at me. Just letting you know.
I also updated the docs link to properly deep link.

> - Cypress now has a targeted Content-Security-Policy and Content-Security-Policy-Report-Only header directive allow list via the [`experimentalCspAllowList`](https://docs.cypress.io/guides/references/configuration#Experimental-Csp-Allow-List) configuration option. Addresses [#1030](https://github.com/cypress-io/cypress/issues/1030). Addressed in [#26483](https://github.com/cypress-io/cypress/pull/26483)

Reviewer 2: ```suggestion
- Cypress can now test pages with targeted `Content-Security-Policy` and `Content-Security-Policy-Report-Only` header directives by specifying the allow list via the [`experimentalCspAllowList`](https://docs.cypress.io/guides/references/configuration#Experimental-Csp-Allow-List) configuration option. Addresses [#1030](https://github.com/cypress-io/cypress/issues/1030).
```

**Features:**

> - Cypress now allows targeted Content-Security-Policy and Content-Security-Policy-Report-Only header directive stripping from requests via the [`stripCspDirectives`](https://docs.cypress.io/guides/references/configuration#stripCspDirectives) config option. Addresses [#1030](https://github.com/cypress-io/cypress/issues/1030). Addressed in [#26483](https://github.com/cypress-io/cypress/pull/26483).

Reviewer 3: Since we've had a couple releases in the interim, can you move this up under a new `12.3.0` heading, release date `05/23/2023`? I promise we'll do everything we can to get this merged in time for that release 😅
